src/main/java/Duke.java

```diff
@@ -1,10 +1,275 @@
+import java.util.Scanner;
+
 public class Duke {
+    private static final String LOGO = " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    private static final String LINE = "-----------------------------";
+    private static final String FIRST_GREETING = "Hello from\n";
+    private static final String SECOND_GREETING = "Hello there! I'm Duke :D";
+    private static final String PROVIDE_SERVICE = "What can I do for you?";
+    private static final String GOODBYE = "Bye! Hope to see you soon :D";
+    private static final String BYE = "bye";
+    private static final String LIST = "list";
+    private static final String MARK = "mark";
+    private static final String MARKED = "Nice! I've marked this task as done:";
+    private static final String UNMARK = "unmark";
+    private static final String UNMARKED = "Nice! I've unmarked this task as undone:";
+    private static final String SPACEBAR = " ";
+    private static final String ADDED = "Got it. I have added this task: ";
+    private static final String NUMBER_OF_TASKS_FIRST_HALF = "Now you have ";
+    private static final String NUMBER_OF_TASKS_SECOND_HALF = " tasks in the list.";
+    private static final String INVALID_INDEX = "The index that you have indicated is invalid, please try again.";
+    private static final String TODO = "todo";
+    private static final String DEADLINE = "deadline";
+    private static final String BY = "/by";
+    private static final String EVENT = "event";
+    private static final String AT = "/at";
+    private static final int START_INDEX_OF_TODO = 5;
+    private static final int START_INDEX_OF_DEADLINE = 9;
+    private static final int START_INDEX_OF_TIMING = 4;
+    private static final int START_INDEX_OF_EVENT = 6;
+    private static final int LENGTH_OF_SPACE = 1;
+    private static final int MINIMUM_INDEX = 0;
+
+    private static Task[] list = new Task[100];
+    private static int index = 0;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printGreeting();
+        processRequest();
+    }
+
+    private static void processRequest() {
+        getRequest();
+    }
+
+    private static void getRequest() {
+        Scanner in = new Scanner(System.in);
+        String input;
+        boolean hasExited = false;
+        while (!hasExited) {
+            input = in.nextLine();
+            input = input.trim();
+            hasExited = filterRequest(input);
+        }
+    }
```

Comment:
I like that you have done refactoring, but I think the getRequest method can be used directly instead of calling processRequest method whose only purpose it to call getRequest.